### PR TITLE
[node] Export TestContext and SuiteContext on v16-v18-v2x

### PR DESCRIPTION
**Pull request by khaosdoctor**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests). _On this one I had V8 stack traces and memory issues, will try to let the bot run the tests_

Select one of these and delete the others:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

URLs:
- https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/67127
- https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/69488

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/v18/test.d.ts
1. types/node/test.d.ts
1. types/node/v18/test/test.ts
1. types/node/test/test.ts
1. types/node/v16/test/test.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cbdcc250d4aeb967591be6e98c7789d63654c2b1
</p>

**Review by ghost — APPROVED** [review]

File: types/node/v16/test/test.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { describe, it, test } from "node:test";
+import { describe, it, test, TestContext } from "node:test";
```
Comment (codershiba) [inline types/node/v16/test/test.ts:1]:
```suggestion
import { describe, it, test, type TestContext } from "node:test";
```

File: types/node/v18/test/test.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Transform, TransformCallback, TransformOptions } from "node:stream";
-import { after, afterEach, before, beforeEach, describe, it, run, test } from "node:test";
+import { after, afterEach, before, beforeEach, describe, it, run, SuiteContext, test, TestContext } from "node:test";
```
Comment (codershiba) [inline types/node/v18/test/test.ts:2]:
```suggestion
import { after, afterEach, before, beforeEach, describe, it, run, type SuiteContext, test, type TestContext } from "node:test";
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/v18/test/test.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7e3f085ef6d8470681cd3ca6d95619ca0276cb57
</p>

**Review by ghost — APPROVED** [review]

**Review by Semigradsky — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

**Comment by khaosdoctor** [comment]
Ready to merge

**Pull request merged**